syft/pkg/cataloger/ai/cataloger_test.go

```diff
@@ -0,0 +1,381 @@
+			},
+			expected: []string{
+				"model1.gguf",
+				"model2.gguf",
+			},
+		},
+		{
+			name: "finds GGUF files in subdirectories",
+			setup: func(t *testing.T) string {
+				dir := t.TempDir()
+				modelsDir := filepath.Join(dir, "models")
+				os.MkdirAll(modelsDir, 0755)
+				createTestGGUFInDir(t, modelsDir, "llama.gguf")
+
+				deepDir := filepath.Join(dir, "deep", "nested", "path")
+				os.MkdirAll(deepDir, 0755)
+				createTestGGUFInDir(t, deepDir, "mistral.gguf")
+
+				return dir
+			},
+			expected: []string{
+				"models/llama.gguf",
+				"deep/nested/path/mistral.gguf",
+			},
+		},
+		{
+			name: "ignores non-GGUF files",
+			setup: func(t *testing.T) string {
+				dir := t.TempDir()
+				createTestGGUFInDir(t, dir, "model.gguf")
+
+				// Create non-GGUF files
+				os.WriteFile(filepath.Join(dir, "readme.txt"), []byte("readme"), 0644)
+				os.WriteFile(filepath.Join(dir, "model.bin"), []byte("binary"), 0644)
+				os.WriteFile(filepath.Join(dir, "config.json"), []byte("{}"), 0644)
+
+				return dir
+			},
+			expected: []string{
+				"model.gguf",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fixtureDir := tt.setup(t)
+
+			pkgtest.NewCatalogTester().
+				FromDirectory(t, fixtureDir).
+				ExpectsResolverContentQueries(tt.expected).
+				TestCataloger(t, NewGGUFCataloger())
+		})
+	}
+}
+
+func TestGGUFCataloger_Integration(t *testing.T) {
+	tests := []struct {
+		name                  string
+		setup                 func(t *testing.T) string
+		expectedPackages      []pkg.Package
+		expectedRelationships []artifact.Relationship
+	}{
+		{
+			name: "catalog single GGUF file",
+			setup: func(t *testing.T) string {
+				dir := t.TempDir()
+				data := newTestGGUFBuilder().
+					withVersion(3).
+					withTensorCount(291).
+					withStringKV("general.architecture", "llama").
+					withStringKV("general.name", "llama3-8b").
+					withStringKV("general.version", "3.0").
+					withStringKV("general.license", "Apache-2.0").
+					withStringKV("general.quantization", "Q4_K_M").
+					withUint64KV("general.parameter_count", 8030000000).
+					build()
+
+				path := filepath.Join(dir, "llama3-8b.gguf")
+				os.WriteFile(path, data, 0644)
+				return dir
+			},
+			expectedPackages: []pkg.Package{
+				{
+					Name:    "llama3-8b",
+					Version: "3.0",
+					Type:    pkg.ModelPkg,
+					Licenses: pkg.NewLicenseSet(
+						pkg.NewLicenseFromFields("Apache-2.0", "", nil),
+					),
+					Metadata: pkg.GGUFFileHeader{
+						ModelFormat:     "gguf",
+						ModelName:       "llama3-8b",
+						ModelVersion:    "3.0",
+						License:         "Apache-2.0",
+						Architecture:    "llama",
+						Quantization:    "Q4_K_M",
+						Parameters:      8030000000,
+						GGUFVersion:     3,
+						TensorCount:     291,
+						Header:          map[string]interface{}{},
+						TruncatedHeader: false,
+					},
+				},
+			},
+			expectedRelationships: nil,
+		},
+		{
+			name: "catalog multiple GGUF files",
```

Comment:
not really a useful test -- this is asserting that the parser function and generic cataloger work well together